galsim/wfirst/wfirst_bandpass.py

```diff
@@ -0,0 +1,124 @@
+    This routine reads in a file containing a list of wavelengths and throughput for all WFIRST
+    bandpasses, and uses the information in the file to create a dictionary.
+
+    In principle it should be possible to replace the version of the file with another one, provided
+    that the format obeys the following rules:
+
+    - There is a column called 'Wave', containing the wavelengths in microns.
+    - The other columns are labeled by the name of the bandpass.
+
+    Currently the bandpasses are not truncated.  We leave it to the user to decide whether they wish
+    to truncate after getting the bandpasses, and the `thin_err` keyword allows the user to choose
+    the relative errors allowed when thinning.
+
+    By default, the routine will set an AB zeropoint using the WFIRST effective diameter and default
+    exposure time.  Setting the zeropoint can be avoided by setting `AB_zeropoint=False`; changing
+    the exposure time that is used for the zeropoint calculation can be used by setting the
+    `exptime` keyword.
+
+    This routine also loads information about sky backgrounds in each filter, to be used by the
+    galsim.wfirst.getSkyLevel() routine.  The sky background information is saved as an attribute in
+    each Bandpass object.
+
+    There are some subtle points related to the filter edges, which seem to depend on the field
+    angle at some level.  This is more important for the grism than for the imaging, so currently
+    this effect is not included in the WFIRST bandpasses in GalSim.
+
+    Example usage
+    -------------
+
+        >>> wfirst_bandpasses = galsim.wfirst.getBandpasses()
+        >>> f184_bp = wfirst_bandpasses['F184']
+
+    @param AB_zeropoint     Should the routine set an AB zeropoint before returning the bandpass?
+                            If False, then it is up to the user to set a zero point.  [default:
+                            True]
+    @param exptime          Exposure time to use for setting the zeropoint; if None, use the default
+                            WFIRST exposure time, taken from galsim.wfirst.exptime.  [default: None]
+    @param thin_err         Relative error allowed when thinning the bandpasses.  [default: 1e-4]
+
+    @returns A dictionary containing bandpasses for all WFIRST filters and grisms.
+    """
+    # Begin by reading in the file containing the info.
+    datafile = os.path.join(galsim.meta_data.share_dir, "afta_throughput.txt")
+    # One line with the column headings, and the rest as a NumPy array.
+    data = np.loadtxt(datafile, skiprows=1).transpose()
+    first_line = open(datafile).readline().rstrip().split()
+    if len(first_line) != data.shape[0]:
+        raise RuntimeError("Inconsistency in number of columns and header line in file %s"%datafile)
+
+    # Identify the index of the column containing the wavelength in microns.  Get the wavelength and
+    # convert to nm.
+    wave_ind = first_line.index('Wave')
+    wave = 1000.*data[wave_ind,:]
```

Comment:
This would be a lot clearer if you read in the data file as
```
data = np.genfromtxt(datafile, names=True)
```
because then you can get the columns by name:
```
wave = 1000. * data['Wave']
```